lib/rbs/inline/ast/annotations.rb

```diff
@@ -509,28 +509,42 @@ def initialize(tree, source)
 
         # `# @rbs module-self [MODULE_SELF]`
         class ModuleSelf < Base
-          attr_reader :constraint #: RBS::AST::Declarations::Module::Self?
+          attr_reader :self_types #: Array[RBS::AST::Declarations::Module::Self]
 
           attr_reader :comment #: String?
 
           # @rbs override
           def initialize(tree, source)
             @tree = tree
             @source = source
+            @self_types = []
 
             module_self = tree.nth_tree!(1)
-            type = module_self.nth_type?(1)
+            self_trees = module_self.non_trivia_trees
+            self_trees.shift
 
-            case type
-            when Types::ClassInstance, Types::Interface
-              @constraint = RBS::AST::Declarations::Module::Self.new(
-                name: type.name,
-                args: type.args,
-                location: nil
-              )
+            while true
+              type = self_trees.shift
+              case type
+              when Types::ClassInstance, Types::Interface
+                @self_types << RBS::AST::Declarations::Module::Self.new(
+                  name: type.name,
+                  args: type.args,
+                  location: nil
+                )
+              end
+
+              break if self_trees.empty?
+
+              next_tree = self_trees.first
+              if next_tree.is_a?(Array) && next_tree[0] == :kCOMMA
+                self_trees.shift
+              else
+                break
+              end
             end
```

Comment:
`module_self.non_trivia_trees.each` with `next`?

Reply:
How about this? We need to check the order of tokens. So it's better to use `#each_slice` instead of `#each`.
But I'm not sure this is better or not.

```
            module_self = tree.nth_tree!(1)
            self_trees = module_self.non_trivia_trees
            self_trees.shift

            self_trees.each_slice(2) do |type, sep|
              case type
              when Types::ClassInstance, Types::Interface
                @self_types << RBS::AST::Declarations::Module::Self.new(
                  name: type.name,
                  args: type.args,
                  location: nil
                )
              else
                # Invalid type found. Ignored.
              end

              if sep.is_a?(Array) && sep[0] == :kCOMMA
                # Separator found. Continue parsing.
              else
                # Trailing comment found. Stop parsing.
                @comment = sep&.to_s
                break
              end
            end
```